jadx-gui/src/main/java/jadx/gui/utils/CodeUsageInfo.java

```diff
@@ -42,17 +47,20 @@ public void processClass(JavaClass javaClass, CodeLinesInfo linesInfo, List<Stri
 
 	private void addUsage(JNode jNode, JavaClass javaClass,
 	                      CodeLinesInfo linesInfo, CodePosition codePosition, List<StringRef> lines) {
-		UsageInfo usageInfo = usageMap.get(jNode);
-		if (usageInfo == null) {
-			usageInfo = new UsageInfo();
-			usageMap.put(jNode, usageInfo);
-		}
+        UsageInfo usageInfo;
+        synchronized (jNode) {
+            usageInfo = usageMap.get(jNode);
```

Comment:
I don't like double synchronization here (one on jNode and one in map)
I suggest using computeIfAbsent method:
```java
UsageInfo usageInfo = usageMap.computeIfAbsent(jNode, key -> new UsageInfo());
```
as description said it is atomic, so it must solve synchronization problem. Anyway, it must be checked.

Reply:
Indeed a nice and clean solution. I wasn't aware of this method, because I usually use Guavas synchronized Multimap.